src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/AsyncMethodDescFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Internal.TypeSystem;
+
+namespace Internal.JitInterface
+{
+    internal class AsyncMethodDescFactory : ConcurrentDictionary<MethodDesc, AsyncMethodDesc>
```

Comment:
Any reason we're using `ConcurrentDictionary` here instead of matching `UnboxingMethodDescFactory` (which uses `Dictionary`)?

Reply:
The async methoddescs should either be short-lived local to JIT interface and use Dictionary; or long-lived global and use ConcurrentDictionary singleton.

The current R2R setup does not make sense to me. It is probably just some missed cleanup. Could you please try to delete R2R copy of UnboxingMethodDescFactory and switch to the shared one?